terraform-mode.el

```diff
@@ -252,18 +252,54 @@
       (when (re-search-forward (concat "/\\(.*?\\)/" provider "\\]") nil t)
         (match-string 1)))))
 
+(defun terraform--get-resource-provider-source (provider &optional dir)
+  "Return provider source for PROVIDER located in DIR."
+  (goto-char (point-min))
+  (let ((source) (file) (file_path) (tf_files))
```

Comment:
Elisp never uses the underscore, traditional style is to use a dash to separate words.

Further, if you are only declaring variables, they can be written in a single paren, `(let (a b c d) ... )`.  They will all be set to nil.

But it's better to try to always declare and initialize at the same time rather than use `setq` later (kind of pretending we have immutability)

Reply:
ok for the first 2 points.

Originally, I did set `source` with `(let ...)`. But the other `*file*` variables can only be set if there are some files to read, which is not the case if the user has just started to write the terraform file and has not yet saved his buffer.

All in all, I found awkward to set `source` in `(let...)`, because it would be the only variable set this way, and it could be overridden later on.

Reply:
On 2nd thought, I've improved the code following your suggestion. Only `tf-files` is not setup in `let`